nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -91,6 +97,18 @@ bool BtActionServer<ActionT>::on_configure()
     node->get_node_waitables_interface(),
     action_name_, std::bind(&BtActionServer<ActionT>::executeCallback, this));
 
+  // Get parameter for monitoring with Groot via ZMQ Publisher
+  node->get_parameter("enable_groot_monitoring", enable_groot_monitoring_);
+  node->get_parameter("groot_zmq_publisher_port", groot_zmq_publisher_port_);
+  node->get_parameter("groot_zmq_server_port", groot_zmq_server_port_);
+
+  // Get parameters for BT timeouts
+  int timeout;
+  node->get_parameter("bt_loop_timeout", timeout);
+  bt_loop_timeout_ = std::chrono::milliseconds(timeout);
+  node->get_parameter("default_server_timeout", timeout);
```

Comment:
Should this be `server_timeout` and `bt_loop_rate` (or `bt_loop_duration`)? The bt loop is used to set the running rate, so I'd think that's the best way to semantically describe it. Because we run at a certain rate, it relates to the timeout, but the timeout isn't it's main purpose (to regulate the BT tick rate). `server_timeout` is what we use in other places in the code base to describe the same thing as well

Reply:
`bt_loop_duration` sounds good. `bt_loop_rate` suggests this parameter takes a frequency value.
`server_timeout` for an individual BT node can be overridden through the BT node input ports. If the user does not set the `server_timeout` input port on a BT node, it uses the default server timeout value that we put on the blackboard. `default_server_timeout` would make semantic sense to describe the actual purpose of this parameter. What do you think?